src/test/java/org/jboss/tm/usertx/TMDelegate.java

```diff
@@ -0,0 +1,227 @@
+package org.jboss.tm.usertx;
+
+import com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionManagerImple;
+import com.arjuna.ats.jbossatx.BaseTransactionManagerDelegate;
+
+import org.jboss.tm.listener.event.TransactionLifecycleTypeNotSupported;
+import org.jboss.tm.listener.event.AssociationLifecycleEventType;
+import org.jboss.tm.listener.event.TransactionLifecycleEventType;
+import org.jboss.tm.listener.event.TransactionLifecycleEvent;
+import org.jboss.tm.listener.event.TransactionLifecycleListener;
+import org.jboss.tm.listener.event.TransactionLifecycleListenerRegistry;
+import org.jboss.tm.listener.event.VetoingTransactionLifecycleListener;
+
+import javax.transaction.HeuristicMixedException;
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.InvalidTransactionException;
+import javax.transaction.RollbackException;
+import javax.transaction.SystemException;
+import javax.transaction.Transaction;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class TMDelegate extends BaseTransactionManagerDelegate implements TransactionLifecycleListenerRegistry {//TransactionListenerRegistry {
```

Comment:
I was going to have BaseTransactionManagerDelegate implement the old and new interfaces but then changed my mind. So comment needs deleting.